src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2310,33 +2306,35 @@ Json::Value NetworkOPsImp::getServerInfo (bool human, bool admin, bool counters)
 
     if (lpClosed)
     {
-        std::uint64_t baseFee = lpClosed->fees().base;
-        std::uint64_t baseRef = lpClosed->fees().units;
+        XRPAmount const baseFee = lpClosed->fees().base;
         Json::Value l (Json::objectValue);
         l[jss::seq] = Json::UInt (lpClosed->info().seq);
         l[jss::hash] = to_string (lpClosed->info().hash);
 
         if (!human)
         {
-            l[jss::base_fee] = Json::Value::UInt (baseFee);
-            l[jss::reserve_base] = Json::Value::UInt (lpClosed->fees().accountReserve(0).drops());
+            l[jss::base_fee] = baseFee.json();
+            l[jss::reserve_base] = lpClosed->fees().accountReserve(0).json();
             l[jss::reserve_inc] =
-                    Json::Value::UInt (lpClosed->fees().increment);
-            l[jss::close_time] =
-                    Json::Value::UInt (lpClosed->info().closeTime.time_since_epoch().count());
+                    lpClosed->fees().increment.json();
+            l[jss::close_time] = Json::Value::UInt (
+                lpClosed->info().closeTime.time_since_epoch().count());
         }
         else
         {
-            l[jss::base_fee_xrp] = static_cast<double> (baseFee) /
-                    SYSTEM_CURRENCY_PARTS;
+            // Make a local specialization of the TaggedFee class, using
+            // XRPAmount as the "unit" to make some of the math here easier.
+            using DropsDouble = units::TaggedFee<XRPAmount, double>;
```

Comment:
It looks like this type is `XRPDouble`, not `DropsDouble`. I also wouldn't use `XRPAmount` as a tag, I'd make another empty struct like the others. I don't think this type is buying us much anyway, and it does seem like we're using this `TaggedFee` type differently than the others. I'd consider just using `double` for this.

Reply:
The reason I'm using `XRPAmount` as a tag is so that the type will be compatible with `XRPAmount` itself using as far as `TaggedFee` is concerned. The reason that works is that `XRPAmount` uses itself as its `unit_type`. https://github.com/ripple/rippled/pull/2910/files/3b93c0eb76d2559a2e33ad2bd3b5e672fad361e5#diff-715593421d046dd34694688c71cfb223R42

What do you think? Overly clever?

Reply:
It's definitely clever (and I mean that as a compliment - I love "cute" solutions like that). But in this case, I don't think we can justify the clever code. Here's my thinking:

1) It's used to convert "drops" into "xrp" (millions of drops). There's no other calculation here, and not likely to be one. I'd probably just use a lambda for this. 

2) A new type that represents xrp (a fundamentally integral type) as a double at least raises my eyebrows a little.

3) We're using the `TaggedFee` type in a different way than we use it in other places (including the way the tag is used). There is joy is reusing a structure in a new way, but it's also easier to understand something if it's always used in roughly the same way.

Your call, of course. But I'd probably lose the `XRPDouble` class.

Reply:
I like your `decimalXRP()` solution for this. Thanks for doing that.